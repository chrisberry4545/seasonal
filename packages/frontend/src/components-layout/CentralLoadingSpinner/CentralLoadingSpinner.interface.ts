export interface ICentralLoadingSpinner {
  isLoading: boolean;
}
