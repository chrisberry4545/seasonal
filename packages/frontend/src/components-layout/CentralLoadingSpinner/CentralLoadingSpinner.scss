@import '../../styles/settings';
@import '../../styles/tools';

.c-central-loading-spinner {
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
  height: calc(100vh - #{$default-top-margin});
}
