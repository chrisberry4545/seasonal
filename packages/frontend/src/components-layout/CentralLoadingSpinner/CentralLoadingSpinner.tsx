import React, { FC } from 'react';

import './CentralLoadingSpinner.scss';

import {
  LoadingSpinner
} from '@chrisb-dev/seasonal-shared-frontend-components';

import {
  ICentralLoadingSpinner
} from './CentralLoadingSpinner.interface';

export const CentralLoadingSpinner: FC<ICentralLoadingSpinner> = ({
  isLoading
}) => (
  isLoading
    ? <div className='c-central-loading-spinner'>
      <LoadingSpinner />
    </div>
    : null
);
