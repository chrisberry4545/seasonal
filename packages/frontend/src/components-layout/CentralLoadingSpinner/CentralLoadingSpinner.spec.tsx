
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { CentralLoadingSpinner } from './CentralLoadingSpinner';

jest.mock('@chrisb-dev/seasonal-shared-frontend-components', () => ({
  LoadingSpinner: () => 'LoadingSpinner'
}));

describe('<CentralLoadingSpinner />', () => {
  let wrapper: ShallowWrapper;

  describe('when is loading', () => {
    beforeEach(() =>
      wrapper = shallow(
        <CentralLoadingSpinner
          isLoading={true} />
      )
    );

    test('renders correctly', () => expect(wrapper).toMatchSnapshot());
  });

  describe('when the not loading', () => {
    beforeEach(() =>
      wrapper = shallow(
        <CentralLoadingSpinner
          isLoading={false} />
      )
    );

    test('does not render anything', () =>
      expect(wrapper.children().exists()).toBe(false));
  });

});
