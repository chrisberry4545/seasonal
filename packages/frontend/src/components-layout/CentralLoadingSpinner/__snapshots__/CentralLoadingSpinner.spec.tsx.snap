// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CentralLoadingSpinner /> when is loading renders correctly 1`] = `
<div
  className="c-central-loading-spinner"
>
  <LoadingSpinner />
</div>
`;
